src/pythia/src/utils/web3.rs

```diff
@@ -0,0 +1,197 @@
+        nonce: Some(nonce),
+        ..Default::default()
+    };
+
+    let signed_tx = w3
+        .accounts()
+        .sign_transaction(tx, from, key_info(), nat::to_u64(chain_id))
+        .await?;
+
+    let tx_hash = retry_until_success!(w3.eth().send_raw_transaction(
+        signed_tx.raw_transaction.clone(),
+        canister::transform_ctx_tx()
+    ))?;
+    web3::wait_for_success_confirmation(&w3, &tx_hash, 60)
+        .await
+        .context(PythiaError::WaitingForSuccessConfirmationFailed)?;
+    Ok(())
+}
+
+pub async fn transfer_all(chain_id: &Nat, to: &str) -> Result<()> {
+    let w3 = instance(chain_id)?;
+    let from = canister::pma().await?;
+    let from_h160 = address::to_h160(&from)?;
+    let to = address::to_h160(to)?;
+
+    let nonce = retry_until_success!(w3.eth().transaction_count(
+        from_h160,
+        None,
+        canister::transform_ctx()
+    ))?;
+    let mut gas_price = retry_until_success!(w3.eth().gas_price(canister::transform_ctx()))?;
+    gas_price = (gas_price / 10) * 12;
+    let mut value =
+        retry_until_success!(w3.eth().balance(from_h160, None, canister::transform_ctx()))?;
+    value -= gas_price * TRANSFER_GAS_LIMIT;
+
+    let tx = TransactionParameters {
+        gas: TRANSFER_GAS_LIMIT.into(),
+        gas_price: Some(gas_price),
+        to: Some(to),
+        value,
+        nonce: Some(nonce),
+        ..Default::default()
+    };
+
+    let signed_tx = w3
+        .accounts()
+        .sign_transaction(tx, from, key_info(), nat::to_u64(chain_id))
+        .await?;
+
+    let tx_hash = retry_until_success!(w3
+        .eth()
+        .send_raw_transaction(signed_tx.raw_transaction.clone(), canister::transform_ctx()))?;
+    web3::wait_for_success_confirmation(&w3, &tx_hash, 60)
+        .await
+        .context(PythiaError::WaitingForSuccessConfirmationFailed)?;
+
+    Ok(())
+}
+
+pub async fn wait_for_success_confirmation<T: Transport>(
+    w3: &Web3<T>,
+    tx_hash: &H256,
+    timeout: u64,
+) -> Result<TransactionReceipt> {
+    let receipt = wait_for_confirmation(w3, tx_hash, timeout).await?;
+
+    let tx_status = receipt.status.expect("tx should be confirmed").as_u64();
+
+    if tx_status != TX_SUCCESS_STATUS {
+        return Err(PythiaError::TxHasFailed.into());
+    }
+
+    Ok(receipt)
+}
+
+pub async fn wait_for_confirmation<T: Transport>(
+    w3: &Web3<T>,
+    tx_hash: &H256,
+    timeout: u64,
+) -> Result<TransactionReceipt> {
+    let call_opts = CallOptionsBuilder::default()
+        .transform(Some(TransformContext {
+            function: TransformFunc(candid::Func {
+                principal: ic_cdk::api::id(),
+                method: "transform".into(),
+            }),
+            context: vec![],
+        }))
+        .cycles(None)
+        .max_resp(None)
+        .build()
+        .expect("failed to build call options");
+
+    let end_time = time::in_seconds() + timeout;
+    while time::in_seconds() < end_time {
```

Comment:
This could be remade as async timer. But you need to find one of it's implementations